### ISSUE TITLE
Missing numbers in the dataset (SoccerNet) causing the low accuracy

### ISSUE THREAD

**Issue by sahil-chaudhary** [mkoshkina/jersey-number-pipeline#4]
Hey,
I was working jersey number identification, and my accuracy turn out to be pretty low assuming the dataset had 99 unique numbers in the jersey and '-1' for non-visible images. 

As it turns our, your model assumes the same for some reason and I got following accuracy for the test dataset in the Soccernet.

![image](https://github.com/user-attachments/assets/7189df63-c631-49de-8161-46d5fc14a65e)

Could you please confirm these results with yours and let me know your assumption?

**Comment by mkoshkina** [comment]
Yes, there are 99 possible jersey number values. In SoccerNet dataset, tracklets that don't have a jersey number visible are assigned the ground truth value of '-1'.
Our model's accuracy is (as reported in the paper) 87.45% for SoccerNet jsersey number test set.  This is the result from using the whole pipeline: main subject filtering, legibility classifier, player pose guided number crop, and then STR model for classification.  Are you getting your results with our model after the whole pipeline?

**Comment by sahil-chaudhary** [comment]
Thanks. There were certain conflicts which lead to this issue.

Also, I would like to mention, the file containing class SAM is named sam2(while cloning using python3 setup.py). This is a very small change from sam.sam import SAM to from sam2.sam import SAM.

**Comment by sahil-chaudhary** [comment]
Also, the inference time for your architecture is pretty high. 11 seconds (approx) for each frame in GTX 1660 which seems a little unreasonable. Are there any possible modifications we can do to achieve faster inference?

**Comment by mkoshkina** [comment]
Thank you for letting me know about the SAM class setup issue. I'll fix that.

In our experiments, we were aiming for accuracy and that led to some sacrifices in efficiency.  We use pose detection to identify the player's torso and then crop and save the images before running STR. Some ideas for speeding up this part would be: 

- instead of writing/reading jersey number crops from disk, feed them directly into STR. 
- replacing the pose detector, which is quite slow, with a jersey number detector. This, however, is not a simple change, as one would need to train/fine-tune a text detector for the task. Out-of-the-box text detectors that we experimented with did not do a good job on jersey number detection.

**Comment by sahil-chaudhary** [comment]
Thank you. I will try some modifications and see if I can somehow improve accuracy by introducing new datasets. I have a weird idea in my mind. Since, pose detection is basically finding the bounding box or the area in the image where the possibility of finding the jersey number is high, we can technically train a small YOLO model (but this would require manual labelling) to detect the same. But this kind of deviates from the original paper. 

If we want to stick to given dataset without further work from our side, this doesn't seem feasible. We might want to go to (better) pose detector.

Let me know if you are still working on this or trying to improve something around it.

Anyway, Thanks a lot. I was working in a project related to my undergraduate course. This kind of helped to move forward.

**Comment by mkoshkina** [comment]
To avoid manual labelling, one idea would be to use pose detector to produce a weakly-labelled dataset and train/fine-tune text detector on that.
I am not currently working on this, though.

Good luck with your project!
